reconcile/gitlab_members.py

```diff
@@ -0,0 +1,126 @@
+import sys
+import logging
+
+import utils.gql as gql
+from utils.config import get_config
+from utils.gitlab_api import GitLabApi
+
+GROUPS_QUERY = """
+{
+    groups: gitlabgroups_v1{
+        managedGroups
+    }
+}
+"""
+
+USERS_QUERY = """
+{
+    users: users_v1{
+        redhat_username
+        roles{
+            permissions{
+                ... on PermissionGitlabGroups_v1{
+                    name
+                    group
+                }
+            }
+        }
+    }
+}
+  """
+
+def get_gitlab_api():
+    config = get_config()
+    gitlab_config = config['gitlab']
+    server = gitlab_config['server']
+    token = gitlab_config['token']
+    return GitLabApi(server, token, ssl_verify=False)
+
+
+def get_groups():
+    gqlapi = gql.get_api()
+    groups = gqlapi.query(GROUPS_QUERY)['groups']
+    group_dict = {}
+    for g in groups[0]['managedGroups']:
+        group_dict[g] = []
+    return group_dict
+
+def get_desired_state():
+    gqlapi = gql.get_api()
+    gl = get_gitlab_api()
+    groups = gqlapi.query(GROUPS_QUERY)['groups']
+    users = gqlapi.query(USERS_QUERY)['users']
+    desired_group_members = get_groups()
+    for g in groups[0]['managedGroups']:
+        for u in users:
+            for r in u['roles']:
+                for p in r['permissions']:
+                    if 'group' in p and p['group'] == g:
```

Comment:
it is strange to me that `service == 'gitlab-group'` is not here...

Reply:
so, in the query you use, those permissions are indeed the only ones returning (with data). the rest of the permissions also return on the query, but they return empty. so we need to filter the empty ones at least.